### PR TITLE
add error message when features are empty

### PR DESCRIPTION
**Pull request by MikeLing**
@karlnapf maybe something like this? :)
screen shot 2017-06-30 at 1 24 50 am

### CODE REVIEW

File: src/shogun/converter/ica/FastICA.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ CFeatures* CFastICA::apply(CFeatures* features)
 	SG_REF(features);
 
 	SGMatrix<float64_t> X = ((CDenseFeatures<float64_t>*)features)->get_feature_matrix();
+	REQUIRE(X.data(), "you are using an empty matrix");
```
Comment (vigsterkr) [inline src/shogun/converter/ica/FastICA.cpp]: it's good but the error message should be rather: "Features have not been provided.\n"

**Comment by vigsterkr** [comment]
one more thing: if you mention in your commit message the issue number on github, github will automagically link it to the PR/commit, so for example you could have a commit message:
`Fix #3884 add error message when features are empty`

**Comment by MikeLing** [comment]
Sorry, I get up late today. I will update this and other pr within today, thank you @vigsterkr ,

**Comment by vigsterkr** [comment]
that's ok, thnx.

**Pull request merged**